### ISSUE TITLE
UnhandledRejection error in `FP.all()`

### ISSUE THREAD

**Issue by justsml** [functional-promises/functional-promises#75]
See Error here: https://repl.it/@justsml/fp-defect-proof#index.js

```js
const FP = require('functional-promises')
const iThrowThings = async () => {
  throw new Error('🔪🔪🔪🔪🔪🔪🔪')
}

const brokenPromises = () => FP.resolve([{ tears: true }])
  .map(iThrowThings)
  .then(() => {
    console.log('whaaaa')
  }).catch(ex => { throw ex })

try {
  brokenPromises()
} catch (ex) {
  console.log('dumb me')
}
```

Throws:

```bash

(node:82) UnhandledPromiseRejectionWarning: Error: 🔪🔪🔪🔪🔪🔪🔪
(node:82) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)
(node:82) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

From @dhigginbotham

**Comment by justsml** [comment]
This works as intended.

[See updated repl.it example](https://repl.it/@justsml/fp-defect-proof#index.js)

TL;DR Calling code must handle any/all rejections with `.catch`
